front/src/stores/globalPreferenceStore.ts

```diff
@@ -26,3 +31,30 @@ export const selectedBaseIdAtom = atom((get) => {
   // This case should not happen except on initial render
   return "0";
 });
+
+/** Persisted atom to cache the value for the Match Product form input in the Box Reconciliation View. */
+export const reconciliationMatchProductAtom = atomWithStorage<ReconciliationMatchProductFields>(
+  "reconciliationMatchProduct",
+  {
```

Comment:
this atom should actually be a array or dictionary with a key being the sourceProductId and the value the prvious selection for this sourceProduct